kernelci.org/content/en/blog/posts/2021/hackfests.md

```diff
@@ -0,0 +1,135 @@
+---
+date: 2021-09-15
+title: "KernelCI Hackfests"
+linkTitle: "KernelCI Hackfests"
+author: Guillaume Tucker
+description: >
+  Lessons learned from the first two hackfests
+---
+
+KernelCI hackfests span over a few days during which a number of contributors
+get together to focus on upstream Linux kernel testing.  So far, mainly kernel
+and automated test system developers have been taking part in the hackfests but
+anyone is welcome to join.  Topics mostly include extending test coverage in
+various ways: enabling new test suites as well as adding test cases to
+established frameworks such as kselftest and LTP, building additional kernel
+flavours, bringing up new types of hardware to be tested...
+
+There have been two hackfests to date. The current plan is to hold one
+every few months.  Future hackfests will be announced on the [KernelCI mailing
+list](https://groups.io/g/kernelci/topics) as well as [LKML](https://lkml.org/)
+and [Twitter](https://twitter.com/kernelci).  Stay tuned!
+
+
+## Connecting the dots
+
+There is a large ecosystem around the Linux kernel which includes testing in
+many shapes and forms: kernel developers, test developers, test system
+developers, OEMs testing fully integrated products...  All these teams of
+people don't necessarily interact with each other very much outside of their
+organisations, and kernel developers aren't necessarily in the habit of writing
+tests as part of their daily work.
+
+Events such as the KernelCI hackfests give a chance for people from these
+different areas to work together on solving common issues and keep the
+ecosystem healthy.  It also helps with shifting the upstream Linux kernel
+development culture towards a more test-driven workflow, to bring mainline
+Linux closer to the real world where it is actually being used.
+
+Let's consider a plausible hackfest story.  A first participant writes a new
+test, say in kselftest.  A second participant enables the test to run in
+KernelCI, which fails in some cases and a kernel bug is found.  A third
+participant makes a fix for the bug, which can then be tested directly in
+KernelCI to confirm it works as expected.  This may even happen on
+[staging.kernelci.org](https://staging.kernelci.org) before the patches for the
```

Comment:
It might be obvious from the name, but I'd add a phrase explaining what staging is for us.

Reply:
Good point, I'll put a link to the documentation page about it.